### DISCUSSION TITLE
How to use XDEBUG?

### DISCUSSION THREAD

**Discussion by SUPERNOVI** [sfx101/docker-stacks#145] [Q&A]
Hi,

I've enabled XDEBUG on my Laravel stack. However I can't for the life of me get it to do anything.
I logged into the terminal and edited 50-xdebug.ini to this and restarted the stack:

```
xdebug.remote_enable=off
xdebug.client_port=9003
xdebug.client_host=localhost
xdebug.idekey=vscode
xdebug.remote_enable=0
xdebug.remote_connect_back=0
xdebug.start_with_request=yes
xdebug.remote_autostart = 1
```

On VSCODE I have a launch config with this:

```
{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Listen for Xdebug",
            "type": "php",
            "request": "launch",
            "hostname": "127.0.0.1",
            "port": 9003
        }
    ]
}
```

I get no errors but my breakpoints do nothing. I've also checked the documentation and can't find any extra information, which is odd considering in the DECK app itself the XDEBUG option is quite prominent.

**Comment by nabad600** [comment]
Hi @SUPERNOVI , 
Thank you for sharing your issue, can you share the Laravel version and PHP version?
We'd be happy to help.

**Comment by SUPERNOVI** [comment]
Hi, appreciate the quick response.

I'm not sure the Laravel version would have any impact, for my testing I'm putting breakpoints in the index.php file, but for reference it's Laravel 9.
PHP version is 8.2.
My OS is Windows 11.

I have a feeling it's something to do with my Windows communicating with the WSL2 environment. However I'm really not sure, happy to follow some guide on how to successfully setup XDEBUG.

**Reply by nabad600** [reply]
Hi @SUPERNOVI, I appreciate your responding. We are working on it and will give you an update soon.

**Comment by nabad600** [comment]
Hi @SUPERNOVI, Thank you for using the deck, We are finding and bug fix is done, Please add extension WSL, PHP debug in your VSCode and add the below line 
![image](https://github.com/sfx101/deck/assets/67466982/9cfae087-efd2-4f98-831d-7f616fa21595)
![image](https://github.com/sfx101/deck/assets/67466982/aebfc4c0-f5d6-4d68-b0fc-1c9d2c4f7e02)
**for reference please look into to below link**
https://www.youtube.com/watch?v=it7JQKPfWTU

Thank you for your input. We'd be glad to assist.

**Reply by SUPERNOVI** [reply]
Thank you, but this has not helped. Most of the instructions in that video are for Docker, which I'm obviously not using.
Can you give me a step by step guide for Deck on what I need to do to get this working?

So do you have to be connected to the WSL for the debug to work? I was under the impression it could remotely work. 
When I connect to the WSL my debug tab is blank, no option to debug nothing. Is this because I need to then install the same extensions on the remote?
It's a little confusing when you start crossing over between Local and Remote (WSL).

**Reply by SUPERNOVI** [reply]
Have you had a chance to read my reply @nabad600 ?
